Find a functionally equivalent code:android.text.Layout.drawBackground:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * @hide
 */

@@ -36,49 +36,43 @@
                             // construction
                             if (mLineBackgroundSpans.spanStarts[j] >= end || mLineBackgroundSpans.spanEnds[j] <= start)
                                 continue;
-                            if (spansLength == spans.length) {
-                                // The spans array needs to be expanded
-                                int newSize = ArrayUtils.idealObjectArraySize(2 * spansLength);
-                                ParagraphStyle[] newSpans = new ParagraphStyle[newSize];
-                                System.arraycopy(spans, 0, newSpans, 0, spansLength);
-                                spans = newSpans;
-                            }
-                            spans[spansLength++] = mLineBackgroundSpans.spans[j];
+                            spans = GrowingArrayUtils.append(spans, spansLength, mLineBackgroundSpans.spans[j]);
+                            spansLength++;
                         }
                     }
                 }

